rust/src/verify.rs

```diff
@@ -526,4 +582,186 @@ mod tests {
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![],
+                    hash: non_empty_spec.hash,
+                },
+                spec: &non_empty_spec,
+                is_true: false,
+                is_left: false,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(spec.empty_child.iter().cloned())
+                        .collect(),
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: true,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(spec.empty_child.iter().cloned())
+                        .collect(),
+                    suffix: vec![0_u8; 32],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: true,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(vec![0_u8; 32].into_iter())
+                        .collect(),
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8]
+                        .into_iter()
+                        .chain(spec.empty_child.iter().cloned())
+                        .chain(b"xxxx".to_vec())
+                        .collect(),
+                    suffix: vec![],
+                    hash: spec.hash,
+                },
+                spec: spec,
+                is_true: false,
+                is_left: true,
+            },
+            EmptyBranchCase {
+                op: ics23::InnerOp {
+                    prefix: vec![1_u8],
+                    suffix: vec![],
+                    hash: non_empty_spec.hash,
+                },
+                spec: &non_empty_spec,
+                is_true: false,
+                is_left: true,
+            },
+        ];
+
+        for (i, case) in cases.iter().enumerate() {
+            let res = if case.is_left {
```

Comment:
Same comments as for the go code. Both to cover both left and right at once as well as to avoid checking invalid ops.